### PR TITLE
[TFLite] Using real image for QNN testing.

### PR DESCRIPTION
**Pull request by anijain2305**
This PR does 2 things
* Test QNN on a real image. QNN and TFLite outputs differ due to rounding differences, making random input not a good candidate for label testing. So, this PR adds a real image for testing QNN.
* @inadob also found a bug in QNN parsing #4807  This PR needs that fix as well to enable the real image tests, so adding that fix here.

Thanks to @inadob for finding the bug

@FrozenGene @kevinthesun @inadob Can you please review?

### CODE REVIEW

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -1124,10 +1124,14 @@ def convert_conv(self, op, conv_type):
...
             do_pad = not (pad_top == 0 and pad_bottom == 0 and pad_left == 0 and pad_right == 0)
             if do_pad:
+                pad_value = 0
+                if input_tensor.qnn_params:
+                    pad_value = get_scalar_from_constant(input_tensor.qnn_params['zero_point'])
                 in_expr = _op.nn.pad(data=in_expr, pad_width=((0, 0),
                                                               (pad_top, pad_bottom),
                                                               (pad_left, pad_right),
-                                                              (0, 0)))
+                                                              (0, 0)), pad_value=float(pad_value))
```
Comment (FrozenGene) [inline python/tvm/relay/frontend/tflite.py]: Must this be float(pad_value)? For QNN, this is int type. Does our pad require it be float type?

Comment (FrozenGene) [reply python/tvm/relay/frontend/tflite.py]: Our convert_pad is not correct too for QNN. We should use OutputZeroPoint for pad_value, please fix it together and write one unit testing for it.

Comment (inadob) [reply python/tvm/relay/frontend/tflite.py]: So before giving the final output, in_expr goes first through the padding and then the result of the _op.pad is passed to _op.conv2d. My concern here is that if we set  pad_value=output_zero_point then wouldn't that be a problem later in _op.conv2d where we do the operation with the input_tensor qnn parameters. Correct me if I am wrong.

Comment (anijain2305) [reply python/tvm/relay/frontend/tflite.py]: Zhao meant just single pad operator, not the pad in conv. I will fix it.

Comment (anijain2305) [reply python/tvm/relay/frontend/tflite.py]: > Must this be float(pad_value)? For QNN, this is int type. Does our pad require it be float type?

Pad only takes float value for now. Internaly it gets converted to the input dtype. So, this is fine for now.

**Comment by anijain2305** [comment]
@inadob Can you please the TFLite pad test portion? I am not very familiar with using fake quant, and I just replicated your code for Pad. So, closer scrutiny would be good.

**Comment by anijain2305** [comment]
Ping @FrozenGene

**Comment by FrozenGene** [comment]
ping @inadob , would you review it again? If you have no other comments, would you set explicit approve? Thanks?

File: tests/python/frontend/tflite/test_forward.py [hunk 2]
```diff
@@ -1100,16 +1117,26 @@ def test_forward_squeeze():
...
-        out = array_ops.pad(in_data[0], ops.convert_to_tensor(data[1], dtype=data[1].dtype))
-        compare_tflite_with_tvm([data[0]], ['in:0'], in_data, [out])
+        in_data = [array_ops.placeholder(shape=data[0].shape, dtype='float32', name='in')]
+
+        if quantized:
+            # fake_quant will keep the tensors in float32 until the conversion in the session
+            inq_data = [tf.quantization.fake_quant_with_min_max_args(in_data[0],
+                                                                     min=-100,
+                                                                     max=100,
+                                                                     name="inq_0")]
```
[the remark this answers is not in the document]
Comment (anijain2305) [reply tests/python/frontend/tflite/test_forward.py]: Thanks @inadob for the review. Will incorporate your comments in few hours.

**Comment by anijain2305** [comment]
@inadob Done

**Comment by anijain2305** [comment]
@FrozenGene Good to merge?

**Comment by FrozenGene** [comment]
Thanks @anijain2305 @inadob It is merged now.

**Comment by masahi** [comment]
What happen if the zero point is a vector, as in per channel quantuzation? What should the pad value be?

**Comment by u99127** [comment]
> 
> 
> What happen if the zero point is a vector, as in per channel quantization? What should the pad value be?

@masahi good question but please file a separate issue only affecting master since per channel quantization came in after this. Otherwise I fear this will be missed. 

Ramana

**Comment by u99127** [comment]
@anijain2305  - this needs a backport to 0.6 tag ?

**Comment by masahi** [comment]
@u99127 I figured the answer myself. QNN support for per channel quantization applies to only weight quantization (which is supposed to be a common restriction in other frameworks as well). For pad value we can use a scalar zero point associated with activations.

**Comment by anijain2305** [comment]
@masahi You are right (as always :) )

@u99127 Let me look into this.

**Pull request merged**